internal/monitors/conviva/accountservice.go

```
@@ -0,0 +1,225 @@
+	client         *httpClient
+	mutex          *sync.RWMutex
+}
+
+// NewAccountService factory function creating AccountService
+func newAccountService(ctx context.Context, timeout *time.Duration, client *httpClient) accountService {
+	service := accountServiceImpl{ctx: ctx, timeout: timeout, client: client, mutex: &sync.RWMutex{},}
+	return &service
+}
+
+func (s *accountServiceImpl) getDefault() *account {
+	s.init()
+	return s.defaultAccount
+}
+
+func (s *accountServiceImpl) getMetricLensDimensionMap(accountName string) map[string]float64 {
+	s.init()
+	for _, a := range s.accounts {
+		if a.Name == accountName {
+			return a.metricLensDimensions
+		}
+	}
+	return nil
+}
+
+func (s *accountServiceImpl) getID(accountName string) string {
+	if len(s.accounts) == 0 {
+		s.init()
+	}
+	for _, a := range s.accounts {
+		if a.Name == accountName {
+			return a.id
+		}
+	}
+	return ""
+}
+
+func (s *accountServiceImpl) getFilters(accountName string) map[string]string {
+	s.init()
+	for _, a := range s.accounts {
+		if a.Name == accountName {
+			return a.filters
+		}
+	}
+	return nil
+}
+
+func (s *accountServiceImpl) getMetricLensFilters(accountName string) map[string]string {
+	s.init()
+	for _, a := range s.accounts {
+		if a.Name == accountName {
+			return a.metricLensFilters
+		}
+	}
+	return nil
+}
+
+func (s *accountServiceImpl) getFilterID(accountName string, filterName string) string {
+	s.init()
+	for _, a := range s.accounts {
+		if a.Name == accountName {
+			for id, name := range a.filters {
+				if name == filterName {
+					return id
+				}
+			}
+		}
+	}
+	return ""
+}
+
+func (s *accountServiceImpl) getMetricLensDimensionID(accountName string, metricLensDimension string) float64 {
+	s.init()
+	for _, a := range s.accounts {
+		if a.Name == accountName {
+			for name, id := range a.metricLensDimensions {
+				if name == metricLensDimension {
+					return id
+				}
+			}
+		}
+	}
+	return 0
+}
+
+func (s *accountServiceImpl) init() {
```

This function should return an error if it couldn't be initialized.